tests/e2e/e2e_tcp_server_first_test.go

```diff
@@ -0,0 +1,83 @@
+package e2e
+
+import (
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/tests/framework"
+	. "github.com/openservicemesh/osm/tests/framework"
+)
+
+var _ = OSMDescribe("TCP server-first traffic",
+	OSMDescribeInfo{
+		Tier:   1,
+		Bucket: 1,
+	},
+	func() {
+		var (
+			sourceNs = framework.RandomNameWithPrefix("client")
+			destNs   = framework.RandomNameWithPrefix("server")
+			ns       = []string{sourceNs, destNs}
+		)
+
+		It("TCP server-first traffic", func() {
+			// Install OSM
+			installOpts := Td.GetOSMInstallOpts()
+			installOpts.EnablePermissiveMode = true
+			Expect(Td.InstallOSM(installOpts)).To(Succeed())
+
+			// Create Test NS
+			for _, n := range ns {
+				Expect(Td.CreateNs(n, nil)).To(Succeed())
+				Expect(Td.AddNsToMesh(true, n)).To(Succeed())
+			}
+
+			destinationPort := 80
+
+			// Get simple pod definitions for the TCP server
+			svcAccDef, podDef, svcDef, err := Td.SimplePodApp(
+				SimplePodAppDef{
+					PodName:     framework.RandomNameWithPrefix("server"),
+					Namespace:   destNs,
+					Image:       "busybox",
+					Command:     []string{"nc", "-lkp", strconv.Itoa(destinationPort), "-e", "sh", "-c", "while yes; do :; done"},
+					Ports:       []int{destinationPort},
+					AppProtocol: constants.ProtocolTCPServerFirst,
+					OS:          Td.ClusterOS,
+				},
+			)
+
+			Expect(err).NotTo(HaveOccurred())
+
+			_, err = Td.CreateServiceAccount(destNs, &svcAccDef)
+			Expect(err).NotTo(HaveOccurred())
+			_, err = Td.CreatePod(destNs, podDef)
+			Expect(err).NotTo(HaveOccurred())
+			dstSvc, err := Td.CreateService(destNs, svcDef)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Expect it to be up and running in it's receiver namespace
+			Expect(Td.WaitForPodsRunningReady(destNs, 120*time.Second, 1, nil)).To(Succeed())
+
+			svcAccDef, podDef, _, err = Td.SimplePodApp(SimplePodAppDef{
+				PodName:   framework.RandomNameWithPrefix("client"),
+				Namespace: sourceNs,
+				Command:   []string{"nc", dstSvc.Name + "." + dstSvc.Namespace, strconv.Itoa(destinationPort)},
```

Comment:
Can you leverage the `TCPRequest` function here which is already cross plat?

Reply:
Yeah, that works here. To make this test fully cross platform though I would need the windows equivalent for the `nc` command on the server side. All it should do is accept a connection and send back some response without reading any of the request. Do you know what that might look like on WIndows?

Reply:
I think swapping it out to `TCPRequest` is better regardless of the cross-platform effort as it increases the consistency among the codebase.

I kind of eluded to one approach with the above comment. If we only want a listener on a specific port Envoy can do this work easily. Where envoy is going to be used as the application container and is going to be sidecared by another envoy.

We know that Envoy is cross-platform and we already have a dependency on it. Also, the project has expertise in envoy so it can be easy to debug.

> Do you know what that might look like on WIndows?


Regarding the equivalent of `nc` on windows, the container that we use won't have that so you will need to do that with PowerShell core. 

If you want to be a good citizen, I would extract the functionality of the server into a function similar to `GetOSSpecificHTTPBinPod` and only implement this for linux and throw a not implemented exception on Windows. That way we have a building block `GetOSSpecificTCPListener` (name up to you) that we can use across all tests

Reply:
Talked with @shashankram offline, I'll work on making this test fully cross-platform later.